src/khoj/processor/conversation/utils.py

```diff
@@ -232,7 +232,8 @@ def truncate_messages(
     tokens = sum([len(encoder.encode(message.content)) for message in messages if type(message.content) == str])
 
     # Drop older messages until under max supported prompt size by model
-    while (tokens + system_message_tokens) > max_prompt_size and len(messages) > 1:
+    # Reserves 4 tokens to demarcate each message (e.g <|im_start|>user, <|im_end|>, <|endoftext|> etc.)
```

Comment:
For all the LLMs? This seems like formatting more specific to a few models.

Reply:
Yeah, this is just a general hack. It'll work for a class of models. For others it will just reduce prompt size by 4 tokens. The main (still unresolved) issue is of the model exceeding it's prompt size limit when responding. Haven't figured out which knob will ensure it doesn't fail with exceeded max prompt size error that it sometimes does for offline chat models